src/library/export/dlgtrackexport.h

```diff
@@ -0,0 +1,62 @@
+#ifndef DLGTRACKEXPORT_H
+#define DLGTRACKEXPORT_H
+
+#include <future>
+
+#include <QDialog>
+#include <QString>
+#include <QScopedPointer>
+#include <QTime>
+
+#include "configobject.h"
+#include "library/export/trackexport.h"
+#include "library/export/ui_dlgtrackexport.h"
+
+// A dialog for interacting with the track exporter in an interactive manner.
+// Handles errors and user interactions.
+class DlgTrackExport : public QDialog,
```

Comment:
I want to make trackexport not depend on a GUI to work, so I don't want it to own the dialog.  This way, I can write tests for the exporter separate from the gui implementation.  What would the interface on the call site look like if the dialog were more dumb?